### PR TITLE
[CI] Move jobs in Travis to Azure DevOps

### PR DESCRIPTION
**Pull request by haroldrandom**
Fix https://github.com/Azure/azure-cli/issues/9329

- Job Dependency Check is duplicated with jobs in AOD, VerifyLinuxRequirements/VerifyWindowsRequirements/VerifyDarwinRequirements , removed,
- Add unit test job for core, telemetry in Python 2.7, 3.6 and 3.8
- Add Integration test against profiles in Python 2.7, 3.6 and 3.8
- Add test for extensions loading in Python 3.6
- Add Job VerifySphinxDocumentGenerator in Python 3.6
- Job Nightly Edge Build is useless, replaced by Edge Build in internal ADO, removed
- Job Automation Docker is useless, removed

for removed job, follow-up is https://github.com/Azure/azure-cli/issues/11510

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by haroldrandom** [comment]
Currently, master branch is vulnerable that can be Squash and merge without CI green.

The job in `.travis.yml` "**_./scripts/ci/reject_master_prs.sh_**" has not taken effect.

This would be dangerous.

**Comment by yungezz** [comment]
> Currently, master branch is vulnerable that can be Squash and merge without CI green.
> 
> The job in `.travis.yml` "**_./scripts/ci/reject_master_prs.sh_**" has not taken effect.
> 
> This would be dangerous.

This could be configured in repo settting. what's the impact of mark CI as required for PR today? If impact is clear, I can do it.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -361,6 +372,54 @@ jobs:
 
     displayName: 'Test pip Install'
 
+- job: UnitTest
+  displayName: Unit Test for Core and Telemetry
+  timeoutInMinutes: 10
+
+  pool:
+    vmImage: 'ubuntu-16.04'
```
Comment (yungezz) [inline azure-pipelines.yml]: why ubuntu only?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -520,6 +579,28 @@ jobs:
...
+  strategy:
+    matrix:
+      Python36:
+        python.version: '3.6'
+  steps:
+    - task: UsePythonVersion@0
+      displayName: 'Use Python $(python.version)'
+      inputs:
+        versionSpec: '$(python.version)'
+    - bash: pip install --upgrade pip wheel
```
Comment (yungezz) [inline azure-pipelines.yml]: curious about this step. this is for test, does that mean user need to do this step when using cli? if yes, this should be in requirements.txt

File: .travis.yml [hunk 3]
```diff
@@ -15,9 +15,6 @@ install: pip install --upgrade pip tox six
 cache: pip
```
[the remark this answers is not in the document]
Comment (haroldrandom) [reply .travis.yml]: By the end of  this Sprint, the last day should be 2019.12.15

File: azure-pipelines.yml (hunk 1 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: I think we should add it in Windows too.

File: azure-pipelines.yml (hunk 2 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: No need for user.
Just an extra step in script `./scripts/ci/test_extensions.sh` that build wheels and install them. So here package `wheel` must be installed ahead.

Actually, this step should be omitted in the future.

File: azure-pipelines.yml (hunk 1 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: Oh, sorry, currently, there is no script supported in Windows platform. All test script are suffix with *.sh. 

So I create another issue to fix that https://github.com/Azure/azure-cli/issues/11450 in S163

**Comment by haroldrandom** [comment]
> > Currently, master branch is vulnerable that can be Squash and merge without CI green.
> > The job in `.travis.yml` "**_./scripts/ci/reject_master_prs.sh_**" has not taken effect.
> > This would be dangerous.
> 
> This could be configured in repo settting. what's the impact of mark CI as required for PR today? If impact is clear, I can do it.

@jiasli suggest that master branch is useless for now, no need to enable this.

File: .travis.yml [hunk 4]
```diff
@@ -1,63 +0,0 @@
...
-install: pip install --upgrade pip tox six
-cache: pip
-jobs:
-  include:
-    - stage: precheck
-      env: PURPOSE='Reject PRs to master'
-      script: ./scripts/ci/reject_master_prs.sh
-    - stage: precheck
-      env: PURPOSE='Dependency Check'
-      script: ./scripts/ci/dependency_check.sh
```
Comment (yungezz) [inline .travis.yml]: where're equivalent of those jobs:  dependency check, test_profile_intergration, automation docker in new azure devops pipeline?

Comment (haroldrandom) [reply .travis.yml]: 1. dependency check -> Jobs: VerifyLinuxRequirements/VerifyWindowsRequirements/VerifyDarwinRequirements
2. test_profile_integration -> Job: IntegrationTestAgainstProfiles
3. Automation docker is useless, the container registry can be found. Current active container registry is managed by @fengzhou-msft

File: azure-pipelines.yml [hunk 5]
```diff
@@ -361,6 +372,54 @@ jobs:
...
+      Python38:
+        python.version: '3.8'
+  steps:
+    - task: UsePythonVersion@0
+      displayName: 'Use Python $(python.version)'
+      inputs:
+        versionSpec: '$(python.version)'
+    - bash: pip install --upgrade pip wheel
+      displayName: 'Install pip and wheel'
+    - bash: ./scripts/ci/test_profile_integration.sh
```
Comment (Juliehzl) [inline azure-pipelines.yml]: It looks what test_profile_integration.sh does is verifying all the modules can be loaded correctly according to [here](https://github.com/Azure/azure-cli/blob/dev/scripts/ci/test_profile_integration.sh#L23). Integration Tests against different profiles are done in job RunAutomationxxxx, right?

File: azure-pipelines.yml [hunk 6]
```diff
@@ -1111,3 +1192,17 @@ jobs:
   - bash: |
       . scripts/ci/install_azdev.sh
       azdev linter --ci-exclusions
+
+- job: VerifySphinxDocumentGenerator
```
Comment (Juliehzl) [inline azure-pipelines.yml]: Do we have document check for extensions CI?

File: azure-pipelines.yml (hunk 5 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: Yes, need to be more accurate.  What name would you like?

File: azure-pipelines.yml [hunk 7]
```diff
@@ -361,6 +372,54 @@ jobs:
...
+- job: IntegrationTestAgainstProfiles
+  displayName: Integration Test against Profiles
+  dependsOn: BuildPythonWheel
+  condition: succeeded()
+  timeoutInMinutes: 20
+
+  pool:
+    vmImage: 'ubuntu-16.04'
+  strategy:
+    matrix:
```
Comment (MyronFanQiu) [inline azure-pipelines.yml]: Do we need 2.7 too?

Comment (haroldrandom) [reply azure-pipelines.yml]: Python 2.7 will get retired by 2020.01

Comment (fengzhou-msft) [reply azure-pipelines.yml]: For awareness, there's a PR scheduled in S163 to drop Python 2.7 support: https://github.com/Azure/azure-cli/pull/11363.

Comment (haroldrandom) [reply azure-pipelines.yml]: Let's Vote!

Comment (haroldrandom) [reply azure-pipelines.yml]: Python 2

Comment (haroldrandom) [reply azure-pipelines.yml]: Python 3

File: azure-pipelines.yml (hunk 6 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: Thanks for remind!
I tried it in CI, it failed if I enable
```
test_commands = [sys.argv[1]] if len(sys.argv) >= 2 else default_commands
```

I have already created a bug issue, will fix it in next Sprint S163 https://github.com/Azure/azure-cli/issues/11495

File: azure-pipelines.yml (hunk 5 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: VerifyCommandsLoadingAgainstProfiles ?

File: azure-pipelines.yml (hunk 7 above)
Comment (yungezz) [reply azure-pipelines.yml]: question, python 2.7 retires at 2020.01, what's our plan on python 2.7 support? this may not related with PR directly.

**Comment by haroldrandom** [comment]
@yungezz @yonzhan @jiasli @zikalino Could one of our azure-cli repo admin disable Travis for this?
https://stackoverflow.com/questions/30874853/how-do-you-remove-a-github-repository-for-travis-ci-tests

File: azure-pipelines.yml (hunk 7 above)
Comment (fengzhou-msft) [reply azure-pipelines.yml]: We're going to drop support for python 2.7 in the next sprint. The plan is to remove tests for python 2.7 and also remove python 2.7 from the classifiers in setup.py. So future code is not guaranteed to be compatible with python 2.7. 

We will not have active support for python 2.7, but anyone from the community could submit PRs to fix python-2.7-breaking code as long as the impact is small.

File: azure-pipelines.yml (hunk 2 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: It's actually requirements.txt

**Pull request merged**